src/dsll/algebra.dsl

```diff
@@ -1,16 +1,18 @@
 tconstructor VectorSpace: type
 tconstructor Vector(v : VectorSpace) : type
 tconstructor Scalar : type
+tconstructor ZeroScalar : type
```

Comment:
why are we distinguishing ZeroScalars as a type?
I would add a predicate `IsZero(c : Scalar) : Prop` if you need that